plugins/out_es/es_conf.c

```diff
@@ -162,9 +162,26 @@ struct flb_elasticsearch *flb_es_conf_create(struct flb_output_instance *ins,
             }
             ctx->aws_region = (char *) tmp;
 
+            tmp = flb_output_get_property("aws_sts_endpoint", ins);
+            if (!tmp) {
+                flb_error("[out_es] aws_sts_endpoint not set");
+                flb_es_conf_destroy(ctx);
+                return NULL;
+            }
+            ctx->aws_sts_endpoint = (char *) tmp;
+
+            tmp = flb_output_get_property("aws_role_arn", ins);
+            if (ctx->aws_sts_endpoint && !tmp) {
+                flb_plg_error(ctx->ins, " 'aws_role_arn' is a required option to use "
```

Comment:
Sorry... I just realized... since we use sts_endpoint in the EKS Provider, this error message is no longer true. Role_arn is not required.

Reply:
Does this mean that role_arn is not required even for the sts provider? I don't fully understand why the error message is not required. Can you explain.

Reply:
Basically there are two cases:
1. role_arn is present, in which case we use the STS Provider which needs both the sts endpoint and the role as arguments
2. role_arn is not present, in which case we use the standard chain provider. That can contain the EKS provider, so it needs the sts endpoint as an argument

Originally, I had forgotten about case #2, so I thought #1 was the only case.

Reply:
ooh! This makes sense! Thank you